app/controllers/api/v1/events_controller.rb

```diff
@@ -28,7 +28,8 @@ def my_events
 
 	def update
 		event = Event.find(params[:id])
-		event.update(
+		if current_user.admin && event.user_id == current_user
+			event.update(
```

Comment:
While you're fixing stuff, I think the tabbing of the items in this `update` call may need to be shifted over one?